### PR TITLE
Handle Path validation inside the PostPolicy handler

### PR DESCRIPTION
**Pull request by krishnasrinivas**
## Description

To handle https://github.com/minio/mint/issues/220

## How Has This Been Tested?

manual

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added unit tests to cover my changes.
- [ ] I have added/updated functional tests in [mint](https://github.com/minio/mint). (If yes, add `mint` PR # here: )
- [ ] All new and existing tests passed.

### CODE REVIEW

File: cmd/bucket-handlers.go [hunk 1]
```diff
@@ -447,6 +448,15 @@ func (api objectAPIHandlers) PostPolicyBucketHandler(w http.ResponseWriter, r *h
 		writeErrorResponse(w, ErrMissingContentLength, r.URL)
 		return
 	}
+	resource, err := getResource(r.URL.Path, r.Host, globalDomainName)
```
Comment (kannappanr) [inline cmd/bucket-handlers.go:450]: you can remove the stale comment on line number 442.

**Review by kannappanr — DISMISSED** [review]
LGTM

File: cmd/bucket-handlers.go [hunk 2]
```diff
@@ -447,6 +447,15 @@ func (api objectAPIHandlers) PostPolicyBucketHandler(w http.ResponseWriter, r *h
...
+	if err != nil {
+		writeErrorResponse(w, ErrInvalidRequest, r.URL)
```
Comment (harshavardhana) [inline cmd/bucket-handlers.go:452]: We should add a return here.. so that we don't move forward.

Comment (krishnasrinivas) [reply cmd/bucket-handlers.go:452]: shucks, good catch.

**Review by kannappanr — APPROVED** [review]
LGTM

**Review by harshavardhana — APPROVED** [review]

**Pull request merged**